### ISSUE TITLE
fw provided mac address fallback fails

### ISSUE THREAD

**Issue by HELLBOY017** [OnePlusOSS/android_kernel_oneplus_sm8250#41]
This works with the kernel shipped with oxygen OS but doesn't work with the public source provided resulting in a different mac address on each reboot. Using `oneplus/sm8250_t_13.1_op8` branch currently. 

Expected behaviour:-
MAC address is matches the device's original mac address and doesn't regenerate on every boot

**Comment by OnePlusOSSAdmin** [comment]
If you find this issue is related to some kernel code or some missing kernel code, please provide detailed information. Then we could check.

**Comment by ppajda** [comment]
MAC address changes every reboot, even if set to device MAC.

**Comment by HELLBOY017** [comment]
> MAC address changes every reboot, even if set to device MAC.

Yeah the mac firmware file path is wrong, fixing it fixed device reading mac address from wlan_mac.bin but when it's empty the mac address is wrong and regenerates on every reboot (This is the correct mac file path for OOS :- https://github.com/HELLBOY017/kernel_oneplus_sm8250/blob/0d3bb2da4c12712ce771588d821541f73f735236/drivers/staging/qcacld-3.0/core/hdd/inc/wlan_hdd_misc.h#L51)

For now I'm using commits from sm8350 to query mac address from dms, it fixes my issue as my wlan_mac.bin is empty.

**Comment by TropsPk** [comment]
@OnePlusOSSAdmin 
@pswbuild

**Comment by TropsPk** [comment]
@OnePlusOSSAdmin waiting for you
